lib/data_stores/mysql_advanced_data_store.py

```diff
@@ -26,23 +26,48 @@ class Error(data_store.Error):
   """Base class for all exceptions in this module."""
 # pylint: enable=nonstandard-exception
 
+class SafeQueue(Queue.Queue):
+  def __init__(self, maxsize=0):
```

Comment:
Unless I don't understand right, if I self.mutex=threading.RLock() before super, then it will just set self.mutex=threading.Lock() and then call these setup functions.
```self.not_empty = _threading.Condition(self.mutex)
self.not_full = _threading.Condition(self.mutex)
self.all_tasks_done = _threading.Condition(self.mutex)```
If I set self.mutex=threading.RLock() after calling super it will have already set those 3 up using a self.mutex using threading.Lock()